From 1673533aa0fbc9324e14f0e73c74767fe511a75b Mon Sep 17 00:00:00 2001
From: Richard Gong <richard.gong@intel.com>
Date: Mon, 26 Apr 2021 09:54:22 -0500
Subject: [PATCH 32/41] HSD #22012990567: crypto: intel_fcs: use the word
 rather than bit

commit 7368ab7874203378c970b2d1d267776f6a292ee0 from
https://github.com/altera-opensource/linux-socfpga.git
branch is socfpga-5.4.114-lts

Using the word rather than bit for certificate to address a comment from
community on upstreamed patch.

Signed-off-by: Richard Gong <richard.gong@intel.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/crypto/intel_fcs.c           | 2 +-
 include/uapi/linux/intel_fcs-ioctl.h | 6 ++----
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/crypto/intel_fcs.c b/drivers/crypto/intel_fcs.c
index f664265ba11f..5765fc5bceb9 100644
--- a/drivers/crypto/intel_fcs.c
+++ b/drivers/crypto/intel_fcs.c
@@ -278,7 +278,7 @@ static long fcs_ioctl(struct file *file, unsigned int cmd,
 		}
 
 		dev_dbg(dev, "Test=%d, Size=%d; Address=0x%p\n",
-			data->com_paras.c_request.test.test_bit,
+			data->com_paras.c_request.test.test_word,
 			data->com_paras.c_request.size,
 			data->com_paras.c_request.addr);
 
diff --git a/include/uapi/linux/intel_fcs-ioctl.h b/include/uapi/linux/intel_fcs-ioctl.h
index 6b8fd3a48904..40963b187f2b 100644
--- a/include/uapi/linux/intel_fcs-ioctl.h
+++ b/include/uapi/linux/intel_fcs-ioctl.h
@@ -33,12 +33,10 @@ enum fcs_certificate_test {
 
 /**
  * struct intel_fcs_cert_test_word - certificate test word
- * @test_bit: if set, do not write fuses, write to cache only.
- * @rsvd: write as 0
+ * @test_word: if set, do not write fuses, write to cache only.
  */
 struct intel_fcs_cert_test_word {
-	uint32_t test_bit:1;
-	uint32_t rsvd:31;
+	uint32_t test_word;
 };
 
 /**
-- 
2.18.1

